src/transformers/models/trocr/processing_trocr.py

```diff
@@ -103,7 +110,10 @@ def from_pretrained(cls, pretrained_model_name_or_path, **kwargs):
                 [`PreTrainedTokenizer`]
         """
         feature_extractor = AutoFeatureExtractor.from_pretrained(pretrained_model_name_or_path, **kwargs)
-        tokenizer = RobertaTokenizer.from_pretrained(pretrained_model_name_or_path, **kwargs)
+        if "trocr-small" in pretrained_model_name_or_path:
+            tokenizer = XLMRobertaTokenizer.from_pretrained(pretrained_model_name_or_path, **kwargs)
+        else:
+            tokenizer = RobertaTokenizer.from_pretrained(pretrained_model_name_or_path, **kwargs)
```

Comment:
```suggestion
tokenizer = AutoTokenizer.from_pretrained(pretrained_model_name_or_path, **kwargs)
```
I think we can replace this by the AutoTokenizer API.